2-structured-data/tests/test_crud.py

```diff
@@ -12,138 +12,115 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import unittest
-
 import bookshelf
 import config
 from flaky import flaky
 from gcloud.exceptions import ServiceUnavailable
-from nose.plugins.attrib import attr
+import py.test
 
 
 def flaky_filter(e, *args):
     return isinstance(e, ServiceUnavailable)
 
 
-@flaky(rerun_filter=flaky_filter)
-class IntegrationBase(unittest.TestCase):
-
-    def createBooks(self, n=1):
-        with self.app.test_request_context():
-            for i in range(1, n + 1):
-                self.model.create({'title': u'Book {0}'.format(i)})
-
-    def setUp(self):
-        self.app = bookshelf.create_app(
-            config,
-            testing=True,
-            config_overrides={
-                'DATA_BACKEND': self.backend
-            }
-        )
-
-        with self.app.app_context():
-            self.model = bookshelf.get_model()
-
-        self.ids_to_delete = []
-
-        # Monkey-patch create so we can track the IDs of every item
-        # created and delete them during tearDown.
-        self.original_create = self.model.create
+@py.test.yield_fixture(params=['datastore', 'cloudsql', 'mongodb'])
+def app(request, monkeypatch):
+    app = bookshelf.create_app(
+        config,
+        testing=True,
+        config_overrides={
+            'DATA_BACKEND': request.param
+        })
 
-        def tracking_create(*args, **kwargs):
-            res = self.original_create(*args, **kwargs)
-            self.ids_to_delete.append(res['id'])
-            return res
+    with app.test_request_context():
+        yield app
 
-        self.model.create = tracking_create
 
-    def tearDown(self):
+@py.test.yield_fixture
+def model(monkeypatch, app):
```

Comment:
Can you add a comment that including 'app' here puts the model in the test_request_context? Seems like that's important, and I can see myself coming back here later and being like "`app` is unused! DELETE"